integrations/GoogleAds/browser.js

```diff
@@ -48,44 +52,95 @@ class GoogleAds {
     logger.debug("===in init Google Ads===");
   }
 
-  identify(rudderElement) {
+  identify() {
     logger.debug("[GoogleAds] identify:: method not supported");
   }
 
   // https://developers.google.com/gtagjs/reference/event
   track(rudderElement) {
     logger.debug("in GoogleAdsAnalyticsManager track");
-    const conversionData = this.getConversionData(
-      this.clickEventConversions,
-      rudderElement.message.event
-    );
-    if (conversionData.conversionLabel) {
-      const { conversionLabel } = conversionData;
-      const { eventName } = conversionData;
-      const sendToValue = `${this.conversionId}/${conversionLabel}`;
-      const properties = {};
+    if (!this.dynamicRemarketing) {
+      const conversionData = this.getConversionData(
+        this.clickEventConversions,
+        rudderElement.message.event
+      );
+      if (conversionData.conversionLabel) {
+        const { conversionLabel } = conversionData;
+        const { eventName } = conversionData;
+        const sendToValue = `${this.conversionId}/${conversionLabel}`;
+        let properties = {};
+        if (rudderElement.message.properties) {
+          properties.value =
+            rudderElement.message.properties.revenue ||
+            rudderElement.message.properties.value;
+          properties.currency = rudderElement.message.properties.currency;
+          properties.transaction_id = rudderElement.message.properties.order_id;
+        }
+        properties.send_to = sendToValue;
+        properties = removeUndefinedAndNullValues(properties);
+        window.gtag("event", eventName, properties);
+      }
+    } else {
+      const { event } = rudderElement.message;
+      if (!event) {
+        logger.error("Event name not present");
+        return;
+      }
+
+      const sendToValue = this.conversionId;
+      let payload = {};
       if (rudderElement.message.properties) {
-        properties.value = rudderElement.message.properties.revenue;
-        properties.currency = rudderElement.message.properties.currency;
-        properties.transaction_id = rudderElement.message.properties.order_id;
+        payload.value =
+          rudderElement.message.properties.revenue ||
+          rudderElement.message.properties.value;
```

Comment:
Why bother mapping this data? Let's set the entire properties object to the payload.

Reply:
Mapping `value` either from `revenue` or `value` as `value` data is generic during dynamic re-marketing and current user does not get mixed with it

Reply:
My assumption is that `revenue` itself is not applicable to remarketing. Anyway, we'll clarify this in the docs for the users.